### ISSUE TITLE
CPU spikes and lockups with Draw buttons inactive state when needed option enabled

### ISSUE THREAD

**Issue by gu1ll0me** [psifidotos/applet-window-buttons#126]
After a certain time of use, when I click the Kicker launcher or the clipboard applet, my CPU utilization goes to 100% for one core and a lockup happens for ~1 second whenever if the Kicker launcher and the clipboard applet are on a plasma panel or a Latte panel. It doesn't happen immediately after Latte starts but only after a couple of minutes or after a certain amount of uses of the widgets. It seems that the more time the system is running, the worse the lockup is.

I tested many things for days and found that the **Draw buttons inactive state when needed** are causing this issue. Without this option, there is no more lockup with Kicker launcher or clipboard applet.

Show/hide Kicker menu + Show/hide clipboard with _Draw buttons inactive state when needed_ enabled after some minutes:
![image](https://user-images.githubusercontent.com/674170/113155210-c9523880-9206-11eb-8b8f-924286a51b9a.png)

Same actions but with _Draw buttons inactive state when needed_ disabled:
![image](https://user-images.githubusercontent.com/674170/113155606-32d24700-9207-11eb-9101-899eb4e06f77.png)

**Config**
ArchLinux
Plasma 5.21.3
Mesa 20.1 (AMD)
Latest Latte-Dock/Window buttons commits
Button theme: Breeze

**Comment by psifidotos** [comment]
What is your copmputer characteristics, cpu, ram, hard disk etc.?

**Comment by psifidotos** [comment]
2. In what process these spikes come from?

**Comment by gu1ll0me** [comment]
> What is your copmputer characteristics, cpu, ram, hard disk etc.?

i7-6700k / 16GB RAM / NVMe SSD / Radeon 6800 XT

>     1. In what process these spikes come from?

From `latte-dock`.
I just created an empty plasma panel to add the window buttons widget to see the result. I will post feedback later.

Thanks.

**Comment by gu1ll0me** [comment]
When the window buttons widget is on an empty plasma panel (without any other widgets), the CPU spikes still occurs but, this time, with the `plasmashell` process. The Kicker menu and clipboard lockups doesn't happens since these widgets stayed in the Latte panel. Each spikes below corresponds to: show the Kicker menu; hide the Kicker menu; show the clipboard widget; hide the clipboard widget.

![image](https://user-images.githubusercontent.com/674170/113521273-6d6d1400-9566-11eb-96c0-3231dc88dc91.png)

As soon as I disable the Draw button inactive state... All come back to normal with any CPU spikes.

Some more config details which may be revelant: I'm on X11 with dual-monitor (3840x2160 + 1920x1200) and I'm using fractional scaling (`xrandr --scale 2x2` for the 4k monitor and `xrandr --scale 1.5x1.5` for the 2k monitor).

**Comment by psifidotos** [comment]
sorry I can not follow and neither I can replicate, but you should be aware about: https://www.reddit.com/r/kde/comments/mj6p31/latte_dock_notice_about_freezes_during_startup/

clipboard is in my black list of applets and I dont use it anymore with Latte.

If you are using latest master, you remove clipboard applet or disable it and everything is fine, then sorry but window buttons applet is not responsible for clipboard applet and application launcher.

I dont have any freezes in my system any more so I suppose this flagges for WorksForMe until there is a concrete way to replicate in my system. The CPU spikes can be normal because any window state change must be tracked down and identified but freezes is important but I can not replicate. I wouldnt be surprised if this AMD driver issue.

**Comment by XEmpty** [comment]
Problem still appear... after some time that applet make my plasma actions delayed. Everything what i wanna do with plasma-shell lagging.

Problem on : 8.1x and 9.x version

**Comment by gu1ll0me** [comment]
@XEmpty Are you using an AMD video card too with the Mesa drivers?

**Comment by psifidotos** [comment]
One thing that you could try is to try another window decoration. There is a chance that Breeze does not play well with that option. You could check the SierraBreezeEnhanced if this is reproducable, I remember that SierraBreezeEnhanced was playing just fine with that option.

You should know that the applet is not drawing the buttons and states for c++ decorations, your chosen decoration is doing it.

For Aurorae themes I believe that this bug is not reproducable.

**Comment by a1d4r** [comment]
Same issue here. 
I have a setup with two monitors, each has a panel with Applet Window Buttons and "Show: Active window is maximized" option selected. When I switch windows on different monitors, CPU utilization reaches 100% and plasma panels are not responsive for a few seconds. CPU temperature reaches 90°C, while the normal temperature is 50-60°C

```
OS: Manjaro Linux x86_64 
Kernel: 5.10.60-1-MANJARO 
Resolution: 2160x1440, 2880x1620 
DE: Plasma 5.22.4 
WM: KWin 
WM Theme: Breeze 
Theme: Breath2 [Plasma], Breath-Dark [GTK2/3] 
Icons: breath2 [Plasma], breath2 [GTK2/3] 
CPU: AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx (8) @ 2.100GHz 
GPU: AMD ATI 03:00.0 Picasso 
Memory: 11343MiB / 14997MiB
```

**Comment by psifidotos** [comment]
this should now fixed with latest master and plasma 5.23.2
